test/tests-regexp-2025.js

```diff
@@ -0,0 +1,17 @@
+if (typeof exports !== "undefined") {
+  var test = require("./driver.js").test
+  var testFail = require("./driver.js").testFail
+}
+
+test("/(?<x>a)|(?<x>b)/", {}, { ecmaVersion: 2025 })
+testFail("/(?<x>a)|(?<x>b)/", "Invalid regular expression: /(?<x>a)|(?<x>b)/: Duplicate capture group name (1:1)", { ecmaVersion: 2024 })
+testFail("/(?<x>a)(?<x>b)/", "Invalid regular expression: /(?<x>a)(?<x>b)/: Duplicate capture group name (1:1)", { ecmaVersion: 2025 })
+test("/(?:(?<x>a)|(?<x>b))\\k<x>/", {}, { ecmaVersion: 2025 })
+testFail("/(?:(?<x>a)|(?<x>b))\\k<x>/", "Invalid regular expression: /(?:(?<x>a)|(?<x>b))\\k<x>/: Duplicate capture group name (1:1)", { ecmaVersion: 2024 })
+testFail("/(?:(?<x>a)(?<x>b))\\k<x>/", "Invalid regular expression: /(?:(?<x>a)(?<x>b))\\k<x>/: Duplicate capture group name (1:1)", { ecmaVersion: 2025 })
+test("/(?<y>a)(?<x>a)|(?<x>b)(?<y>b)/", {}, { ecmaVersion: 2025 })
+test("/(?<x>a)|(?<x>b)|(?<x>c)/", {}, { ecmaVersion: 2025 })
+test("/(?<x>a)|\\k<x>/", {}, { ecmaVersion: 2025 })
+testFail("/(?<x>a)|(?<x>b)(?<x>c)/", "Invalid regular expression: /(?<x>a)|(?<x>b)(?<x>c)/: Duplicate capture group name (1:1)", { ecmaVersion: 2025 })
+testFail("/(?:(?<x>a)|(?<x>b))(?<x>c)/", "Invalid regular expression: /(?:(?<x>a)|(?<x>b))(?<x>c)/: Duplicate capture group name (1:1)", { ecmaVersion: 2025 })
```

Comment:
When I try it in Safari, it's not a syntax error. I may be misunderstanding something 🤔

Reply:
If I try the swapped `/(?<x>c)(?:(?<x>a)|(?<x>b))/` in Safari, I will get a syntax error.
Is it a bug in Safari that `/(?:(?<x>a)|(?<x>b))(?<x>c)/` does not result in a syntax error? 🤔

Reply:
The way I read the spec, yes, that should be a syntax error—there is no `|` parse node that separates either of the first two definitions of `<x>` from the last one.

Reply:
Thank you for your comment!
I was relieved that my understanding was correct 😄
If the current specs are fine, I think this PR is ready for review.